BSP/Simulator/Src/BSP_SPI.c

```diff
@@ -29,7 +32,24 @@ void BSP_SPI_Init(spi_port_t port, bsp_os_t *spi_os, bool baremetal) {
  * @return  None
  */
 void BSP_SPI_Write(spi_port_t port, uint8_t *txBuf, uint32_t txLen) {
-    // TODO
+    char* spi = port ? "SPI3 LTC2315" : "SPI LTC6811";
+    char* msg;
+    if(!initialized) {
+        asprintf(&msg, "Used %s without initialization!\n", spi);
+        Simulator_Log_Location(LOG_ERROR, msg);
+        free(msg);
+        Fault_BitMap |= Fault_CRC;
+        EnterFaultState();
```

Comment:
I think officially we are supposed to post the Fault semaphore but for sim i dont think it really matters

i did the same last sem and it worked for the 1 test case we were debugging then